cmake/flash/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+get_property(ENV_VARS GLOBAL PROPERTY FLASH_SCRIPT_ENV_VARS)
+list(APPEND ENV_VARS
+  O=${PROJECT_BINARY_DIR}
+  KERNEL_ELF_NAME=${KERNEL_ELF_NAME}
+  KERNEL_HEX_NAME=${KERNEL_HEX_NAME}
+  KERNEL_BIN_NAME=${KERNEL_BIN_NAME}
+  )
+
+if(FLASH_SCRIPT)
+  add_custom_target(flash
```

Comment:
the flash target needs to depend on generated binary, i.e. I need to be able to run 'make flash' without having a binary ready first (it will build the binary then flash)

Reply:
@nashif : Fixed